grapple/types/redirects.py

```diff
@@ -1,38 +1,76 @@
+import copy
+
+from typing import Optional
+
 import graphene
 
-from django.conf import settings
 from wagtail.contrib.redirects.models import Redirect
+from wagtail.models import Page, Site
+
+from grapple.types.sites import SiteObjectType
 
 from .pages import get_page_interface
 
 
-class RedirectType(graphene.ObjectType):
+class RedirectObjectType(graphene.ObjectType):
     old_path = graphene.String(required=True)
     old_url = graphene.String(required=True)
-    new_url = graphene.String(required=True)
+    new_url = graphene.String(required=False)
     page = graphene.Field(get_page_interface())
+    site = graphene.Field(SiteObjectType, required=False)
```

Comment:
It could be, though I'm not sure if that would be clear for users given that `site` is not required for `redirects`. Maybe throwing a custom exception instead would be better?

Reply:
We're not saying it's required for `Redirect` instances though, we're saying it's required when querying for them via GraphQL (which it is as `RedirectsQuery` makes sure of that)